posthog/api/insight.py

```diff
@@ -0,0 +1,33 @@
+from typing import Any, Dict
+
+from django.db.models import QuerySet
+from rest_framework import serializers, viewsets
+
+from posthog.models import Insight
+
+
+class InsightSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Insight
+        fields = ["id", "name", "filters", "created_at", "pinned"]
+
+    def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Insight:
+        request = self.context["request"]
+        annotation = Insight.objects.create(team=request.user.team_set.get(), **validated_data)
+        return annotation
+
+
+class InsightViewSet(viewsets.ModelViewSet):
```

Comment:
yeah i think it could work. the one problem i could anticipate with this pattern is at some point down the line when we have 20 different possible insights we realize it actually would make sense creating models of each one like we have for models because the filter object becomes a huge blob of params that's different for each insight type

also, regarding this PR, if you want to extend this endpoint should i continue implementing with it. I just spent a few minutes unrooting it to adopt the dashboard_item endpoint